surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProvider.java

```diff
@@ -104,22 +104,29 @@ public JUnitPlatformProvider( ProviderParameters parameters )
     public RunResult invoke( Object forkTestSet )
                     throws TestSetFailedException, ReporterException
     {
+        ReporterFactory reporterFactory = parameters.getReporterFactory();
+        RunListener runListener = reporterFactory.createReporter();
+
+        ConsoleOutputCapture.startCapture( (ConsoleOutputReceiver) runListener );
+
         if ( forkTestSet instanceof TestsToRun )
         {
-            return invokeAllTests( (TestsToRun) forkTestSet );
+            invokeAllTests( (TestsToRun) forkTestSet, runListener );
         }
         else if ( forkTestSet instanceof Class )
         {
-            return invokeAllTests( TestsToRun.fromClass( (Class<?>) forkTestSet ) );
+            invokeAllTests( TestsToRun.fromClass( (Class<?>) forkTestSet ), runListener );
         }
         else if ( forkTestSet == null )
         {
-            return invokeAllTests( scanClasspath() );
+            invokeAllTests( scanClasspath(), runListener );
         }
         else
         {
             throw new IllegalArgumentException( "Unexpected value of forkTestSet: " + forkTestSet );
         }
+
+        return reporterFactory.close();
```

Comment:
Why you do not call it in `finally` block as before?

Reply:
Wouldn't it swallow the `IllegalArgumentException`?

Reply:
@sormuras 
Don't break the meaning and behavior of previous code. The `close` is not your problem. The problem was with method `scanClasspath` and not the `close` method.